File: Data Structures & Algorithms/validate-parentheses/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
		let ref = new Map();
		let stk = [];
		stk.length = 0;
		ref.set( ')', '(' );
		ref.set( '}', '{' );
		ref.set( ']', '[' );
		console.log( ref.get( ']' ) )
		for ( const ch of s ) {
			if ( ref.has( ch ) ) {
				if ( stk.length === 0 || ref.get( ch ) != stk[ stk.length - 1 ] ) {
					return false;
				}
				else {
					stk.pop();
				}
			}

			else stk.push( ch );
		}

		return stk.length === 0;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
#include <charconv>
class Solution {
public:
    int evalRPN(vector<string> &tokens)
{
	if (0 == tokens.size())
		return 0;
	else
	{
		int res = 0;
		stack<int> stk;
		for (int i = 0; i < (int)tokens.size(); i++)
		{
			int x;
			if (tokens[i] != "*" && tokens[i] != "+" && tokens[i] != "-" && tokens[i] != "/")
			{
				std::from_chars(tokens[i].data(), tokens[i].data() + (int)tokens[i].size(), x);
				stk.push(x);
			}
			else
			{
				int temp2 = stk.top();
				stk.pop();
				int temp1 = stk.top();
				stk.pop();
				switch (tokens[i][0])
				{
				case 42:
					stk.push(temp1 * temp2);
					break;
				case 43:
					stk.push(temp1 + temp2);
					break;
				case 45:
					stk.push(temp1 - temp2);
					break;
				case 47:
					stk.push(temp1 / temp2);
					break;
				default:
					break;
				}
			}
		}
		return stk.top();
	}
	return 0;
}
};
